src/lib/comms/include/sol-netctl.h

```diff
@@ -76,6 +76,95 @@ typedef struct sol_netctl_network_params {
 } sol_netctl_network_params;
 
 /**
+ * @enum sol_netctl_agent_error_type
+ *
+ * @brief agent error type
+ *
+ * When the network connection failure,
+ * One of these must be chosen to return for connection.
+ * According to the return type, the failure network
+ * connection will be tried again or not.
+ */
+enum sol_netctl_agent_error_type {
+    /*
+     * @brief the error type is retry
+     *
+     * When the network connection failure,
+     * the user can select the return type
+     * and the connection will be tried again.
+     */
+    SOL_NETCTL_AGENT_RETRY,
+    /*
+     * @brief the error type is noretry
+     *
+     * When the network connection failure,
+     * the user can select the return type
+     * and the connection will not be tried again.
+     */
+    SOL_NETCTL_AGENT_NORETRY,
+};
+
+/**
+ * @enum sol_netctl_agent_prompt_type
+ *
+ * @brief agent prompt type
+ *
+ * When the user need input information to login the connecting network,
+ * One of these must be chosen to return the right login type for connection.
+ * According to the return type, the login process will be done.
+ */
+enum sol_netctl_agent_prompt_type {
```

Comment:
I'm wondering if it's more scalable and easier if we use this as a string (as done for technology/type).